x-pack/plugins/session_view/public/components/SessionViewDetailPanel/helpers.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { Process, ProcessFields } from '../../../common/types/process_tree';
+import { DetailPanelProcess } from '../../types';
+
+const getDetailPanelProcessLeader = (leader: ProcessFields) => ({
+  ...leader,
+  id: leader.entity_id,
+  entryMetaType: leader.entry_meta?.type || '',
+  userName: leader.user.name,
+  entryMetaSourceIp: leader.entry_meta?.source.ip || '',
+});
+
+export const getDetailPanelProcess = (process: Process | null) => {
+  const processData = {} as DetailPanelProcess;
+  if (!process) {
+    return processData;
+  }
+
+  processData.id = process.id;
+  processData.start = process.events[0]?.['@timestamp'] || '';
+  processData.end = process.events[process.events.length - 1]?.['@timestamp'] || '';
+  const args = new Set<string>();
+  processData.executable = [];
+
+  process.events.forEach((event) => {
+    if (!processData.user) {
+      processData.user = event.user.name;
+    }
+    if (!processData.pid) {
+      processData.pid = event.process.pid;
+    }
+
+    if (event.process.args.length > 0) {
+      args.add(event.process.args.join(' '));
+    }
+    if (event.process.executable) {
+      processData.executable.push([event.process.executable, `(${event.event.action})`]);
+    }
+    if (event.process.exit_code) {
+      processData.exit_code = event.process.exit_code;
+    }
+  });
+
+  processData.args = [...args];
+  processData.entryLeader = getDetailPanelProcessLeader(process.events[0].process.entry_leader);
```

Comment:
I missed the update on `getDetailPanelProcess` when the UX was changed from having no default `selectedProcess` to having the session leader to be the default `selectedProcess`. The optional chaining in `getDetailPanelProcess` can be removed now since it does not need to expect a null process anymore. We won't need optional chaining 
for `process.events[0]` since the `process` was constructed from grouping `events` by each `process`.

Thanks for catching it @michaelolo24! I will push a fix to this.